### DISCUSSION TITLE
Slow wakeup time after idle for a while

### DISCUSSION THREAD

**Discussion by xproda** [freeplane/freeplane#1525] [Help]
After being idle for a while, Freeplane seems to need tens of seconds to wake up. It affects the thinking flow and productivity. Can it be faster?

**Comment by i-plasm** [comment]
Hello, 

what do you mean by "idle", your computer being idle or just Freeplane while you are working on some other application? What do you exactly mean by "seconds to wake up"? Does the issue happen with very simple maps as well, or only with maps with particular characteristics (for instance very large maps, or maps with significant usage of clones or conditional styles or formulas? Do you use any "init" scripts? Do you use more than one instance of Freeplane simultaneously? What's your OS, OS version, type of Freeplane (.exe, .app, portable, 'bin', FlatPak, some sort of custom launching, etc)? What's the version of Freeplane?

**Comment by xproda** [comment]
1. On "idle", I mean just Freeplane being idle while I'm working on the
other apps of the computer. The idle time can be more than an hour.
2. I usually open a lot of big maps (could be 20~30 maps and each map could
be up to 1000 nodes, but none or few images).
2-1. Recently I intentionally reduced it to ~5 maps, the response was
faster, but still needed a few seconds. I compared it to a commercial
mindmap app at the same computer and at the same conditions and that one
needed only about 1~2 seconds. BTW, I like Freeplane very much and have
used it for more than 10 years, so I truly hope Freeplane can overcome this
issue.
3. No conditional styles. No formulas. No init scripts. Only one instance.
4. Os: Win 11. Freeplane 1.11.8 (1.11.7 for earlier test)

Thanks.

bbarbosa ***@***.***> 於 2023年11月15日 週三 下午11:36寫道：

**Comment by i-plasm** [comment]
Thousands of nodes should not be an issue the way you describe your usage, at least in my experience. Does the same happen with older versions of freeplane or only after a particular version? Maybe you can try giving it more memory. You can set it it to ` -Xmx2G ` or even ` -Xmx4G `. If you launch Freeplane via the .exe set it on the freeplane.l4j.ini file, or if you launch it via the .bat file then on the .bat itself (or set it on both to cover all bases). You can find those files in the freeplane installation directory.

**Comment by xproda** [comment]
Thanks, I'll check it out.

Best regards,
Charles

bbarbosa ***@***.***> 於 2023年11月17日 週五 下午10:12寫道：

**Comment by dpolivaev** [comment]
I think that your computer might have not enough RAM. Therefore, while Freeplane is not used, the RAM it originally allocated could be moved to the SWAP file on your hard drive so that other apps can use it. It would explain why waking up feels very slow.

Nowadays depending on how you use your computer, 16 to 32 MB RAM might be needed. Anything less than that is likely to make troubles, but depending on the other software you use it might be not enough.

Also, if some other app on your computer has a memory leak or generally wants to consume a lot of memory, it could let the memory occupied by other apps including Freeplane be put into the SWAP file.

**Comment by xproda** [comment]
Thanks for the tips. I have changed the -Xmx setting from -Xmx512m to
-Xmx2G. And it seems to improve a lot. Thanks.

Dimitry Polivaev ***@***.***> 於 2023年11月18日 週六 下午5:03寫道：
